src/clr-angular/utils/focus/key-focus/interfaces.ts

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+export interface FocusableItem {
+  tabIndex: number;
+  focus: () => void;
+  nativeElement?: HTMLElement;
```

Comment:
The focus directive does not have to use a specific type of child directive or element. This would allow it to work with public APIs without having to change the public API directive as long as it implements the `FocusItem` interface. If there is no `nativeElement` reference it assumes that it is receiving a list of native DOM elments.